### DISCUSSION TITLE
LOKI: Query editor "line limit" setting to be able to use dashboard variable

### DISCUSSION THREAD

**Discussion by Natreg** [grafana/grafana#49390] [Dashboards, Panels & Visualizations]
On Loki, It would be a good idea to provide a way to use a dashboard variable on the panel json setting "maxLines" ("line limit" on query editor).

This way, users that are just viewers could have an option to use some presets defined by an admin in order to show more lines if their computer/browser allows it.

This will should never exceed the datasource limit though. And in case it does, it would still default to the max value on the datasource configuration.

Right now, using a dashboard variable there just gives an error when the parser tries to convert the text into an int.

"strconv.Atoi: parsing "$totalLines": invalid syntax"

this works very well for other data on the json like the datasources, or the intervals.

**Comment by R-Studio** [comment]
Any news on this?

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
